File: src/main/java/me/mattstudios/triumphpets/listeners/PetListener.java
package me.mattstudios.triumphpets.listeners;

import me.mattstudios.triumphpets.TriumphPets;
import org.bukkit.entity.EntityType;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageEvent;
import org.bukkit.event.entity.EntityTargetLivingEntityEvent;

public class PetListener implements Listener {

    private TriumphPets plugin;

    public PetListener(TriumphPets plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    public void onWolfTargetPet(EntityTargetLivingEntityEvent event) {
        if (!event.getEntityType().equals(EntityType.WOLF)) return;
        if (event.getTarget() == null) return;
        if (plugin.getPetController().isPet(event.getTarget())) event.setCancelled(true);
    }

    @EventHandler
    public void onPetDamage(EntityDamageEvent event) {
        if (plugin.getPetController().isPet(event.getEntity())) event.setCancelled(true);
    }

}

File: src/main/kotlin/me/mattstudios/triumphpets/listeners/PetListener.kt
package me.mattstudios.triumphpets.listeners

import me.mattstudios.triumphpets.TriumphPets
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageEvent
import org.bukkit.event.entity.EntityTargetLivingEntityEvent


/**
 * @author Matt
 */
class PetListener(private val plugin: TriumphPets) : Listener {

    /**
     * Makes sure entities don't target the pet
     */
    @EventHandler
    fun EntityTargetLivingEntityEvent.onEntityTargetPet() {
        val target = target ?: return
        isCancelled = plugin.petManager.petController.isPet(target)
    }

    /**
     * Makes pet invincible
     */
    @EventHandler
    fun EntityDamageEvent.onPetDamage() {
        isCancelled = plugin.petManager.petController.isPet(entity)
    }

}